packages/story-editor/src/app/taxonomy/taxonomyProvider.js

```diff
@@ -191,11 +198,29 @@ function TaxonomyProvider(props) {
     (taxonomy, termSlugs = []) =>
       setSelectedSlugs((selected) => ({
         ...selected,
-        [taxonomy.rest_base]: termSlugs,
+        [taxonomy.rest_base]:
+          typeof termSlugs === 'function'
+            ? termSlugs(selected[taxonomy.rest_base])
+            : termSlugs,
       })),
     []
   );
 
+  // Fetch hierarchical taxonomies on mount
+  useEffect(() => {
+    // only fetch when `shouldRefetchCategories` is true
+    if (shouldRefetchCategories && taxonomies?.length) {
+      const hierarchicalTaxonomies = taxonomies.filter(
+        (taxonomy) => taxonomy.hierarchical
+      );
+      hierarchicalTaxonomies.forEach((taxonomy) =>
+        addSearchResultsToCache(taxonomy, { perPage: 100 })
+      );
+
+      setShouldRefetchCategories(false);
+    }
+  }, [addSearchResultsToCache, shouldRefetchCategories, taxonomies]);
```

Comment:
What is happening here? Just get 100 terms. What if they taxonomy has more than 100?

Reply:
yeaup. that's the current plan of action. I think having  >100 categories in a taxonomy won't be a frequent scenario for users, but that being said, I'm not sure. If it is, we should add that behavior in a separate ticket to not bloat this PR.

If you have any sample interactions of what Gutenberg does with > 100 categories, we'd love to see it. Seemed kind of cumbersome to manually enter in >100 new category terms on any environment. Happy to create a ticket for this and prioritize accordingly here with that info.

@swissspidy @spacedmonkey @BrittanyIRL any thoughts on how often of a use case this might be and the priority on it?

Reply:
Kinda seems like something we can add in to the ability to check and uncheck categories, since this PR is just to add them in.

Reply:
Added a new issue for this so we can do this separately! https://github.com/google/web-stories-wp/issues/9056

Reply:
> Seemed kind of cumbersome to manually enter in >100 new category terms on any environment.

FWIW you could use WP-CLI to easily create thousands of terms easily by simply running `wp term generate web-story-category --count=1000`